docs/how-to-build-with-ddn/with-mongodb.mdx

```diff
@@ -0,0 +1,399 @@
+
+Additionally, we'll familiarize you with the steps and workflows necessary to iterate on your API.
+
+This tutorial assumes you're starting from scratch. We'll use a locally running MongoDB instance via Docker and connect
+it to Hasura, but you can easily follow the steps if you already have data seeded; Hasura will never modify your source
+schema.
+
+<Prereqs />
+
+:::info Additionally, you'll need mongosh
+
+This tutorial uses mongosh — the Mongo shell — to interact with the local MongoDB instance. You can download it
+[here](https://www.mongodb.com/try/download/shell).
+
+:::
+
+## Tutorial
+
+### Step 1. Authenticate your CLI
+
+```sh title="Before you can create a new Hasura DDN project, you need to authenticate your CLI:"
+ddn auth login
+```
+
+This will launch a browser window prompting you to log in or sign up for Hasura DDN. After you log in, the CLI will
+acknowledge your login, giving you access to Hasura Cloud resources.
+
+### Step 2. Scaffold out a new local project
+
+```sh title="Next, create a new local project:"
+ddn supergraph init my-project && cd my-project
+```
+
+Once you move into this directory, you'll see your project scaffolded out for you. You can view the structure by either
+running `ls` in your terminal, or by opening the directory in your preferred editor.
+
+### Step 3. Initialize your MongoDB connector
+
+```sh title="In your project directory, run:"
+ddn connector init my_mongo -i
+```
+
+From the dropdown, start typing `mongo` and hit enter to accept the default port. Then, provide the following connection
+string:
+
+```plaintext
+mongodb://local.hasura.dev:27017/my_database
+```
+
+### Step 4. Start the local MongoDB container
+
+```sh title="Begin by creating a compose file for the Mongo service:"
+touch app/connector/my_mongo/compose.mongo.yaml
+```
+
+```yaml title="Then, open the file and add the following:"
+services:
+  mongodb:
+    image: mongo:latest
+    container_name: mongodb
+    ports:
+      - "27017:27017"
+```
+
+```sh title="Run the container:"
+docker compose -f app/connector/my_mongo/compose.mongo.yaml up -d
+```
+
+```sh title="Use the mongosh shell to seed the database:"
+docker exec -it mongodb mongosh --eval "
+db = db.getSiblingDB('my_database');
+db.users.insertMany([
+  { user_id: 1, name: 'Alice', age: 25 },
+  { user_id: 2, name: 'Bob', age: 30 },
+  { user_id: 3, name: 'Charlie', age: 35 }
+]);
+"
+```
+
+The shell will return information about the newly-inserted `users` records.
+
+### Step 6. Introspect your MongoDB database
+
+```sh title="Next, use the CLI to introspect your MongoDB database:"
+ddn connector introspect my_mongo
+```
+
+After running this, you should see a representation of your collection's schema in
+`app/connector/my_mongo/schema/users.json`; you can view this using `cat` or open the file in your editor.
+
+For each collection in your database, the MongoDB connector will generate a separate JSON file representing it.
+
+### Step 7. Add your model
```

Comment:
For now I think we need to restart the connector between steps 6 & 7. This shouldn't be necessary after we implement the "print schema" capability in the mongo connector. I've added a subtask to ENG-1270 to update this tutorial once that is implemented.

Reply:
Here, we don't have the connector running yet.

Reply:
Oh right - good point